### ISSUE TITLE
Flake: Deleting pod: something-cgroup-conmon: EBUSY

### ISSUE THREAD

**Issue by edsantiago** [containers/podman#14057]
Yep, jinxed myself yesterday.
```
# podman [options] play kube --replace /tmp/podman_test1113683552/kube.yaml --configmap /tmp/podman_test1113683552/foo-cm.yaml
time="2022-04-28T17:02:07Z" level=error msg="Deleting pod 6eeba76dfdf7c4ab35cd5e7064e286b43371aa36d8630ecdca4ef08181f6a3c6 cgroup /libpod_parent/6eeba76dfdf7c4ab35cd5e7064e286b43371aa36d8630ecdca4ef08181f6a3c6: remove /sys/fs/cgroup/libpod_parent/6eeba76dfdf7c4ab35cd5e7064e286b43371aa36d8630ecdca4ef08181f6a3c6/conmon: remove /sys/fs/cgroup/libpod_parent/6eeba76dfdf7c4ab35cd5e7064e286b43371aa36d8630ecdca4ef08181f6a3c6/conmon: device or resource busy"
Error: error removing pod 6eeba76dfdf7c4ab35cd5e7064e286b43371aa36d8630ecdca4ef08181f6a3c6 conmon cgroup: remove /sys/fs/cgroup/libpod_parent/6eeba76dfdf7c4ab35cd5e7064e286b43371aa36d8630ecdca4ef08181f6a3c6/conmon: remove /sys/fs/cgroup/libpod_parent/6eeba76dfdf7c4ab35cd5e7064e286b43371aa36d8630ecdca4ef08181f6a3c6/conmon: device or resource busy
```
Looks similar to #11946 but different enough to merit a new issue.

Seen just now in [int f36 root](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4607778919546880/html/int-podman-fedora-36-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and-replace-should-reuse-the-configmap-volume--1) on an in-flight PR; logs show it happening since April 2:

### Podman play kube [It] podman play kube test env value from configmap and --replace should reuse the configmap volume 
* fedora-35 : int podman fedora-35 root container
  * PR #14051
    * [04-28 08:19](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4973817507348480/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--1)
    * [04-28 08:19](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4973817507348480/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--2)
    * [04-28 08:19](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4973817507348480/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--3)
  * PR #13955
    * [04-21 15:02](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5529787685928960/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--1)
    * [04-21 15:02](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5529787685928960/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--2)
    * [04-21 15:02](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5529787685928960/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--3)
  * PR #13865
    * [04-14 15:26](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5008658164613120/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--1)
    * [04-14 15:26](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5008658164613120/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--2)
    * [04-14 15:26](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5008658164613120/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--3)
  * PR #13811
    * [04-08 10:03](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5926478717124608/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--1)
    * [04-08 10:03](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5926478717124608/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--2)
    * [04-08 10:03](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5926478717124608/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--3)
  * PR #13808
    * [04-22 12:24](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5205147130265600/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--1)
    * [04-22 12:24](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5205147130265600/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--2)
    * [04-22 12:24](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5205147130265600/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--3)
  * PR #13796
    * [04-07 00:06](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5770276460822528/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--1)
    * [04-07 00:06](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5770276460822528/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--2)
    * [04-07 00:06](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5770276460822528/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--3)
 * PR #13757
    * [04-02 17:00](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6393619794886656/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--1)
    * [04-02 17:00](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6393619794886656/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--2)
    * [04-02 17:00](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6393619794886656/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-test-env-value-from-configmap-and---replace-should-reuse-the-configmap-volume--3)
 
 
### Podman play kube [It] podman play kube teardown 
* fedora-34 : int podman fedora-34 root container
  * PR #13249
    * [02-16 10:48](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5391338182017024/html/int-podman-fedora-34-root-container.log.html#t--podman-play-kube-teardown--1)
* fedora-35 : int podman fedora-35 root container
  * PR #14051
    * [04-28 08:19](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4973817507348480/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-teardown--1)
  * PR #13955
    * [04-21 15:02](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5529787685928960/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-teardown--1)
  * PR #13808
    * [04-22 12:24](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5205147130265600/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-teardown--1)
  * PR #13255
    * [02-17 13:25](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4536459939020800/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-teardown--1)
    * [02-17 13:25](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4536459939020800/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-teardown--2)
    * [02-17 13:25](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4536459939020800/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-teardown--3)

**Comment by giuseppe** [comment]
not sure it is the right fix, since I wasn't able to reproduce locally, but there is one race condition I am aware of in the pod cleanup code: https://github.com/containers/podman/pull/14061

**Comment by vrothberg** [comment]
@edsantiago have you seen it flake after #14061?

**Comment by edsantiago** [comment]
No, but only three PRs have merged since then.

**Comment by edsantiago** [comment]
Yep, [still happening](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/6686950391283712/html/int-podman-fedora-36-root-container.log.html) (f36 root, PR is based on very-latest main)

**Comment by edsantiago** [comment]
[f35 root too](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/4540703693864960/html/int-podman-fedora-35-root-container.log.html#t--podman-play-kube-replace--1), in a different test

**Comment by giuseppe** [comment]
let's also kill the processes contained in the cgroup, in the same way as the systemd backend does: https://github.com/containers/common/pull/1019

A bigger hammer, but I think it solves the race condition we are observing

**Comment by rhatdan** [comment]
When containers/common gets merged into Podman we can close this issue.
